python/lsst/obs/ctio0m9/ingest.py

```diff
@@ -70,18 +72,24 @@ def translate_imgType(self, md):
         if val in conversion:
             return conversion[val]
         else:
+            self.log.warn('Unknown image type %s found in IMAGETYP key', val)
             return 'NOT_TRANSLATED'
 
     def translate_wavelength(self, md):
-        """Get the monochromator wavelength from the header, for flats only.
+        """Get the illumination wavelength.
+
+        Get the monochromator wavelength from the header, for flats only.
         Method will need ammending if/when this value is written elsewhere
         """
         val = md.get("OBJECT").rstrip().lstrip()
         if self.translate_imgType(md) != 'flat':
             return float('nan') # defaults to NaN if not a flat
-        if val[0:3].isdigit():
+        if val[0:4].isdigit():
+            wavelength = float(val[0:4])
+        elif val[0:3].isdigit():
             wavelength = float(val[0:3])
-            if wavelength<300: #blindly using the first 3 digits, so if >=1000nm could be bad
+            if wavelength<300 or wavelength>1150: #We don't know what might be stored here,
+                                                  #so a little sanity checking is good
```

Comment:
I don't think those need defining - they're properties of the band-gap of silicon, and so not only will not change, but this is for a specific obs_package, and so won't ever be upgraded to use detectors outside the range.